pkg/kvstore/client.go

```diff
@@ -20,8 +20,14 @@ import (
 	log "github.com/Sirupsen/logrus"
 )
 
-// Client is the instance of the key-value as configured
-var Client KVClient
+var (
+	// Client is the instance of the key-value as configured
+	Client KVClient
```

Comment:
Don't export this. Instead, define a Client() accessor function, to make sure this is not overwritten accidentally.
I'd use sync.Once.Do in that Client() function, to ensure that this is a singleton.